core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -370,161 +285,81 @@ public final class Flags {
-                // fallthrough
-            default:
-                if (type == null) {
-                    logger.info("Using nio");
-                    type = TransportType.NIO;
-                }
-                break;
-        }
-        TRANSPORT_TYPE = type;
-    }
+    private static final boolean USE_LEGACY_ROUTE_DECORATOR_ORDERING =
+            getValue(FlagsProvider::useLegacyRouteDecoratorOrdering, "useLegacyRouteDecoratorOrdering");
 
-    private static final Path DEFAULT_MULTIPART_UPLOADS_LOCATION;
+    private static final boolean USE_DEFAULT_SOCKET_OPTIONS =
+            getValue(FlagsProvider::useDefaultSocketOptions, "useDefaultSocketOptions");
 
-    static {
-        DEFAULT_MULTIPART_UPLOADS_LOCATION = Paths.get(get("defaultMultipartUploadsLocation",
-                                                           System.getProperty("java.io.tmpdir") +
-                                                           File.separatorChar + "armeria" +
-                                                           File.separatorChar + "multipart-uploads",
-                                                           ignore -> true));
-    }
+    private static final boolean ALLOW_DOUBLE_DOTS_IN_QUERY_STRING =
+            getValue(FlagsProvider::allowDoubleDotsInQueryString, "allowDoubleDotsInQueryString");
 
-    /**
-     * Returns the {@link Sampler} that determines whether to retain the stack trace of the exceptions
-     * that are thrown frequently by Armeria.
-     *
-     * @see #verboseExceptionSamplerSpec()
-     */
-    public static Sampler<Class<? extends Throwable>> verboseExceptionSampler() {
-        return VERBOSE_EXCEPTION_SAMPLER;
-    }
+    private static final Path DEFAULT_MULTIPART_UPLOADS_LOCATION =
+            getValue(FlagsProvider::defaultMultipartUploadsLocation, "defaultMultipartUploadsLocation");
 
     /**
-     * Returns the specification string of the {@link Sampler} that determines whether to retain the stack
+     * Returns the specification of the {@link Sampler} that determines whether to retain the stack
      * trace of the exceptions that are thrown frequently by Armeria. A sampled exception will have the stack
      * trace while the others will have an empty stack trace to eliminate the cost of capturing the stack
      * trace.
      *
-     * <p>The default value of this flag is {@value #DEFAULT_VERBOSE_EXCEPTION_SAMPLER_SPEC}, which retains
-     * the stack trace of the exceptions at the maximum rate of 10 exceptions/sec.
+     * <p>The default value of this flag is {@value DefaultFlagsProvider#VERBOSE_EXCEPTION_SAMPLER},
+     * which retains the stack trace of the exceptions at the maximum rate of 10 exceptions/sec.
      * Specify the {@code -Dcom.linecorp.armeria.verboseExceptions=<specification>} JVM option to override
      * the default. See {@link Sampler#of(String)} for the specification string format.</p>
      */
-    public static String verboseExceptionSamplerSpec() {
```

Comment:
Even though it is not used internally, I don't want to make a breaking change.
Could we just return the value of the JVM option `-Dcom.linecorp.armeria.verboseExceptions` and mark it deprecated?

Reply:
Ok, Got it. Thanks